### PR TITLE
Positive semidefinite rank of a matrix

### LINKED ISSUE #646: Compute the positive semidefinite rank of a nonnegative matrix
For a nonnegative matrix, $M$, the *positive semidefinite rank* (PSD rank) of $M$ is the smallest integer $k$ for which there exist PSD matrices $A_i$ and $B_j$ of size $k \times k$ such that $M_{i,j} = \text{Tr}(A_i B_j)$. 

The term is defined here: 
https://arxiv.org/pdf/1407.4095

As an example, for the following matrix from Equation (21) of [arXiv:2406.16566](https://arxiv.org/pdf/2406.16566),

$$
A = 1/2
\begin{pmatrix}
    0 & 1 & 1 \\
    1 & 0 & 1 \\
    1 & 1 & 0
\end{pmatrix}
$$

The PSD rank of this matrix is equal to $2$.

$$
\text{rank}_{\text{PSD}}(A) = 2
$$

### PR DESCRIPTION
**Pull request by vprusso**
Closes #646

### CODE REVIEW

File: toqito/matrix_props/positive_semidefinite_rank.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+import cvxpy as cp
+import numpy as np
+from toqito.matrix_props import is_square
+
+
+def positive_semidefinite_rank(mat: np.ndarray, max_rank: int = 10) -> int | None:
+    r"""Compute the positive semidefinite rank (PSD rank) of a nonnegative matrix.
+
+    The definition of PSD rank is defined in :cite:`Fawzi_2015_Positive`.
```
Comment (purva-thakre) [inline toqito/matrix_props/positive_semidefinite_rank.py:11]: image

image


These show up as a blank entries in the RTD build. I think this is due to the `bib` entry not having a URL in the description. 

https://toqito--647.org.readthedocs.build/en/647/autoapi/matrix_props/positive_semidefinite_rank/index.html

File: toqito/matrix_props/positive_semidefinite_rank.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    return None
+
+
+def _check_psd_rank(mat: np.ndarray, k: int) -> bool:
+    """Check if the given PSD rank k is feasible for matrix M.
+
+    :param mat: 2D numpy ndarray
+    :param max_rank: The maximum rank to check.
+    :return: True if k is a feasible PSD rank, False otherwise.
+    """
```
Comment (purva-thakre) [inline toqito/matrix_props/positive_semidefinite_rank.py:63]: input arg labels and the `param` labels are inconsistent. Choose `mat, max_rank` or `M, k`.

**Review by purva-thakre — APPROVED** [review]
LGTM except for some minor things!

File: toqito/matrix_props/positive_semidefinite_rank.py [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+import numpy as np
+from toqito.matrix_props import is_square
+
+
+def positive_semidefinite_rank(mat: np.ndarray, max_rank: int = 10) -> int | None:
+    r"""Compute the positive semidefinite rank (PSD rank) of a nonnegative matrix.
+
+    The definition of PSD rank is defined in :cite:`Fawzi_2015_Positive`.
+
+    Finds the PSD rank of matrix M by checking feasibility for increasing k.
```
[the remark this answers is not in the document]
Comment (vprusso) [reply toqito/matrix_props/positive_semidefinite_rank.py]: Changed this to `max_rank`. Good catch!

File: toqito/matrix_props/positive_semidefinite_rank.py (hunk 1 above)
Comment (vprusso) [reply toqito/matrix_props/positive_semidefinite_rank.py:12]: Fixed!

File: toqito/matrix_props/positive_semidefinite_rank.py (hunk 2 above)
Comment (vprusso) [reply toqito/matrix_props/positive_semidefinite_rank.py]: Made consistent.

**Pull request merged**